esp-hal/src/spi/mod.rs

```diff
@@ -98,14 +98,16 @@ pub enum BitOrder {
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 #[instability::unstable]
 pub enum DataMode {
-    /// `Single` Data Mode - 1 bit, 2 wires.
+    /// Clock, CS and one data line (SIO0)
+    SingleThreeWire,
+    /// `Single` Data Mode - 1 bit, two data lines. (SIO0, SIO1)
     Single,
-    /// `Dual` Data Mode - 2 bit, 2 wires
+    /// `Dual` Data Mode - 2 bits, two data lines. (SIO0, SIO1)
     Dual,
-    /// `Quad` Data Mode - 4 bit, 4 wires
+    /// `Quad` Data Mode - 4 bit, 4 data lines. (SIO0 .. SIO3)
     Quad,
     #[cfg(spi_octal)]
-    /// `Octal` Data Mode - 8 bit, 8 wires
+    /// `Octal` Data Mode - 8 bit, 8 data lines. (SIO0 .. SIO7)
     Octal,
 }
```

Comment:
Wild thought here with regards to naming consistency.
I think it actually makes more sense for `Single` to mean "3 wire half duplex".
This way, `Single`, `Dual`, `Quad` and `Octal` all mean half duplex and the name strictly refers to the number of data pins used.
Then there can be a special mode for when there are two data lines that aren't used at the same time.

Reply:
Makes sense - and did that. Maybe the now `Single` variant should be named `ThreeWire` to make things more obvious? I _think_ three-wire and four-wire are more common names

Reply:
My issue with the "FourWire" name is that both Dual and Quad use four wires and it'll be somewhat unclear that this means "Single" but over the full duplex wiring.

Reply:
yeah - I'm open to naming suggestions

Reply:
Some inspiration

- `HalfAndHalf`
- `DualHalves`
- `Couple`
- `Half`
- `TwoHalves`
- `SingleSplit`
- `Split`
- `Pair`
- `Serial`
- `SingleWithTwoWires`
- `Single(SingleMode::ThreeWire)`